src/code/z_play.c

```diff
@@ -110,34 +110,254 @@
+    gSaveContext.respawn[0].tempCollectFlags = globalCtx->actorCtx.flags.collectible[2];
+    globalCtx->nextEntranceIndex = gSaveContext.respawn[0].entranceIndex;
+    gSaveContext.respawnFlag = 1;
+    func_80169ECC(globalCtx);
+    globalCtx->sceneLoadFlag = 0x14;
+    globalCtx->unk_1887F = 2;
+}
+
+// Gameplay_LoadToLastEntrance ?
+// Used by game_over and Test7
+void func_80169F78(GameState* gameState) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    globalCtx->nextEntranceIndex = gSaveContext.respawn[2].entranceIndex;
+    gSaveContext.respawnFlag = -1;
+    func_80169ECC(globalCtx);
+    globalCtx->sceneLoadFlag = 0x14;
+    globalCtx->unk_1887F = 2;
+}
+
+// Gameplay_TriggerRespawn ?
+// Used for void by Wallmaster, Deku Shrine doors. Also used by Player, Kaleido, DoorWarp1
+void func_80169FDC(GameState* gameState) {
+    func_80169F78(gameState);
+}
+
+// Used by Kankyo to determine how to change the lighting, e.g. for game over.
+s32 func_80169FFC(GameState* gameState) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    return globalCtx->roomCtx.currRoom.mesh->type0.type != 1;
+}
+
+s32 FrameAdvance_IsEnabled(GameState* gameState) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+
+    return globalCtx->frameAdvCtx.enabled != 0;
+}
+
+// Unused, unchanged from OoT, which uses it only in one Camera function.
+/**
+ * @brief Tests if \p actor is a door and the sides are different rooms.
+ *
+ * @param[in] gameState GameState, promoted to globalCtx inside.
+ * @param[in] actor Actor to test.
+ * @param[out] yaw Facing angle of the actor, or reverse if in the back room.
+ * @return true if \p actor is a door and the sides are in different rooms, false otherwise
+ */
+s32 func_8016A02C(GameState* gameState, Actor* actor, s16* yaw) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+    TransitionActorEntry* transitionActor;
+    s8 frontRoom;
+
+    if (actor->category != ACTORCAT_DOOR) {
+        return false;
+    }
+
+    transitionActor = &globalCtx->doorCtx.transitionActorList[(u16)actor->params >> 10];
+    frontRoom = transitionActor->sides[0].room;
+    if (frontRoom == transitionActor->sides[1].room) {
+        return false;
+    }
+
+    if (frontRoom == actor->room) {
+        *yaw = actor->shape.rot.y;
+    } else {
+        *yaw = actor->shape.rot.y + 0x8000;
+    }
+
+    return true;
+}
+
+// Unused
+/**
+ * @brief Tests if \p pos is underwater.
+ *
+ * @param[in] globalCtx GlobalContext
+ * @param[in] pos position to test
+ * @return True if inside a waterbox and not above a void.
+ */
+s32 Play_IsUnderwater(GlobalContext* globalCtx, Vec3f* pos) {
+    WaterBox* waterBox;
+    CollisionPoly* poly;
+    Vec3f waterSurfacePos;
+    s32 bgId;
+
+    waterSurfacePos = *pos;
+
+    // == true required to match.
+    if ((WaterBox_GetSurface1(globalCtx, &globalCtx->colCtx, waterSurfacePos.x, waterSurfacePos.z, &waterSurfacePos.y,
```

Comment:
```suggestion
    WaterBox* waterBox;
    CollisionPoly* poly;
    Vec3f waterSurfacePos  = *pos
    s32 bgId;

    if ((WaterBox_GetSurface1(globalCtx, &globalCtx->colCtx, waterSurfacePos.x, waterSurfacePos.z, &waterSurfacePos.y,
```
It might be just me. But I think it's obvious that `== true` is not equivalent to `!= 0` and that it's obvious from looking at the code

Reply:
Oh yeah, I spuriously equated it with `== false`.